sunpy/map/maputils.py

```diff
@@ -1,7 +1,309 @@
+
+        """
+
+        if coordinate:
+            if not isinstance(coordinate[0], (SkyCoord, frames.BaseCoordinateFrame)):
+                raise ValueError('coordinate needs to be an Astropy coordinate frame or SkyCoord instance.')
+
+            meta = _get_meta_from_coordinate(coordinate[0])
+
+        else:
+            print('Using default meta assuming Earth based observer at current time with `arcsec` cunit type')
+            meta = _default_meta()
+
+        # if no crval options set, default to 0, 0
+        if crval is None:
+            meta['crval1'], meta['crval2'] = 0, 0
+        else:
+            meta['crval1'], meta['crval2'] = crval[0], crval[1]
+
+        # if no crpix options set, default to 0, 0
+        if crpix is None:
+            meta['crpix1'], meta['crpix2'] = (data.shape[0] + 1)/2., (data.shape[1] + 1)/2.
+        else:
+            meta['crpix1'], meta['crpix2'] = crpix[0], crpix[1]
+
+        # if no cdelt options set, default to 1, 1
+        if cdelt is None:
+            meta['cdelt1'], meta['cdelt2'] = 1, 1
+        else:
+            meta['cdelt1'], meta['cdelt2'] = cdelt[0], cdelt[1]
+
+        meta_dict = MetaDict({
+                    'ctype1': meta.get('ctype1'),
+                    'ctype2': meta.get('ctype2'),
+                    'cunit1': meta.get('cunit1'),
+                    'cunit2': meta.get('cunit1'),
+                    'crpix1': meta.get('crpix1'),
+                    'crpix2': meta.get('crpix2'),
+                    'cdelt1': meta.get('cdelt1'),
+                    'cdelt2': meta.get('cdelt2'),
+                    'crval1': meta.get('crval1'),
+                    'crval2': meta.get('crval2'),
+                    'hgln_obs': meta.get('hgln_obs'),
+                    'hglt_obs': meta.get('hglt_obs'),
+                    'date_obs': meta.get('date_obs'),
+                    't_obs': meta.get('t_obs'),
+                    'dsun_obs': meta.get('dsun_obs'),
+                    'rsun_obs': meta.get('rsun_obs'),
+                    'rsun_ref': meta.get('rsun_ref')
+        })
+
+        for key in kwargs:
+            if key in acceptable_meta:
+                meta_dict[key] = kwargs[key]
+
+        return meta_dict
+
+
+def _default_meta():
+    """
+    Creates default meta data if none is specified
+    """
+    coord_meta = {}
+
+    coord_meta['ctype1'], coord_meta['ctype2'] = 'HPLN-TAN', 'HPLT-TAN'
+    coord_meta['cunit1'], coord_meta['cunit2'] = 'arcsec', 'arcsec'
+    coord_meta['hgln_obs'], coord_meta['hglt_obs'] = 0.0, 0.0
+    coord_meta['dsun_obs'] = con.au.to(u.m).value
+    coord_meta['rsun_ref'] = con.radius.to(u.m).value
+    coord_meta['rsun_obs'] = ((con.radius / con.au).decompose() * u.radian).to(u.arcsec).value
+    coord_meta['t_obs'] = parse_time('now').iso
+    coord_meta['date_obs'] = parse_time('now').iso
+
+    return coord_meta
+
+
+def _get_meta_from_coordinate(coordinate):
+    """
+    Function to get WCS meta data from a SkyCoord or frame using
```

Comment:
```suggestion
    Function to get WCS meta data from a `~astropy.coordinates.SkyCoord` or frame using
```

Not sure if that is right.